File: src/main/java/kr/co/lunatalk/domain/product/domain/ProductColor.java
package kr.co.lunatalk.domain.product.domain;

import jakarta.persistence.*;
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ProductColor extends BaseTimeEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "product_id")
	private Product product;

	private String color;


	@Builder(access = AccessLevel.PRIVATE)
	public ProductColor(Product product, String color) {
		this.product = product;
		this.color = color;
	}

	public static ProductColor createProductColor(Product product, String color) {
		return ProductColor.builder()
			.product(product)
			.color(color)
			.build();
	}

	public void updateProductColor(String color) {
		this.color = color;
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/product/domain/ProductColor.kt
package kr.co.lunatalk.domain.product.domain

import jakarta.persistence.*
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity

@Entity
open class ProductColor protected constructor() : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    open var product: Product? = null
        protected set

    open var color: String? = null
        protected set

    private constructor(product: Product, color: String) : this() {
        this.product = product
        this.color = color
    }

    fun updateProductColor(color: String) {
        this.color = color
    }

    companion object {
        fun createProductColor(product: Product, color: String): ProductColor {
            return ProductColor(product, color)
        }
    }
}
